mitodl/open-discussions
# fix course search textbox behavior

#### Pre-Flight checklist


#### What are the relevant tickets?

no ticket, but basically to accomplish the same thing I did here: https://github.com/mitodl/hugo-course-publisher/pull/190 for the OCW search.

#### What's this PR do?

this removes the automatic search behavior, so that the user has to manually trigger the search instead via the enter key.

I also found a few performance issues on the page and diagnosed those, adding `React.memo` to fix.

also refactored `CourseSearchPage` from class-based to functional component.

#### How should this be manually tested?

This touched a lot of parts of the search, so please test it exhaustively. Facets should all work, should be able to change text search, clear facet values, etc etc.
